pkg/object/globalfilter/globalfilter.go

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package globalfilter
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/megaease/easegress/pkg/context"
+	"github.com/megaease/easegress/pkg/object/httppipeline"
+	"github.com/megaease/easegress/pkg/supervisor"
+	"github.com/megaease/easegress/pkg/util/yamltool"
+)
+
+const (
+	// Category is the category of GlobalFilter.
+	Category = supervisor.CategoryBusinessController
+
+	// Kind is the kind of GlobalFilter.
+	Kind = "GlobalFilter"
+
+	beforePipelineKey = "before"
+	afterPipelineKey  = "after"
+)
+
+type (
+	// GlobalFilter is a business controller
+	// provide handler before and after pipeline in HTTPServer
+	GlobalFilter struct {
+		super     *supervisor.Supervisor
+		superSpec *supervisor.Spec
+		spec      *Spec
+
+		// pipelines map[string]*httppipeline.HTTPPipeline
+		// only contains two key: before and after
+		pipelines *sync.Map
```

Comment:
As there are only two possible keys, propose to use two `sync.Value` instead of `sync.Map`, this is also to align with the `Spec` definition below, where `BeforePipeline` and `AfterPipeline` are used instead of a map.

Reply:
ok, i get it.